blog/2022-04-06-multi-cluster-management.md

```diff
@@ -0,0 +1,285 @@
+
+### Before Starting
+
+1. Prepare a Kubernetes cluster as the control plane of KubeVela.
+1. Make sure [KubeVela v1.3] (https://github.com/tag-dev/KubeVela/releases/tag/v1.3.0) and KubeVela CLI v1.3.0 have been installed successfully.
+2. The list of Kubeconfig from sub clusters that you want to manage. We will take three clusters naming beijing-1, beijing-2 and us-west-1 as examples.
+3. Download and combine with [Multi-Cluster-Demo] (https://github.com/oam-dev/sample/tree/master/12.multi_cluster_demo) to better understand how to use the KubeVela multi-cluster capabilities.
+
+### Distribute to Multiple Specified Clusters
+Distributing multiple specified clusters is the most basic multi-cluster management operation. In KubeVela, you will use a policy called `topology` to implement it. The cluster will be listed in the attribute `clusters`, an array.
+
+First let's make sure switching kubeconfig to the control plane cluster, go with `vela cluster join` to include in the 3 clusters of Beijing-1, Beijing-2 and us-west-1:
+```
+➜   vela cluster join beijing-1.kubeconfig --name beijing-1
+➜   vela cluster join beijing-2.kubeconfig --name beijing-2
+➜   vela cluster join us-west-1.kubeconfig --name us-west-1
+➜   vela cluster list
+CLUSTER        	TYPE           	ENDPOINT                 	ACCEPTED	LABELS
+beijing-1      	X509Certificate	https://47.95.22.71:6443 	true
+beijing-2      	X509Certificate	https://47.93.117.83:6443	true
+us-west-1      	X509Certificate	https://47.88.31.118:6443	true
+```
+Then open multi-cluster-demo, look into `Basic.yaml`:
+```
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: example-app
+  namespace: default
+spec:
+  components:
+    - name: hello-world-server
+      type: webservice
+      properties:
+        image: crccheck/hello-world
+        port: 8000
+      traits:
+        - type: scaler
+          properties:
+            replicas: 3
+        - type: gateway
+          properties:
+            domain: testsvc-mc.example.com
+            # classInSpec : true   If the sub clusters has Kubernetes versions below v1.20 installed, please add this field
+            http:
+              "/": 8000
+  policies:
+    - type: topology
+      name: beijing-clusters
+      properties:
+        clusters: ["beijing-1","beijing-2"]
+```
+It can be seen that this app uses the component of type `webservice` and distributes 3 Deployments to beijing-1 and beijing-2 clusters through the `topology` policy.
+
+**Please note that the premise of successfully distributing resource into sub-cluster is that it must exist a exactly the same namespace as control plane did. Since each cluster has the **`default`** namespace by default, we won't be worry in this case. But suppose we change the namespace in **`basic.yaml`** to be **`multi-cluster`**, we will receive an error: **
```

Comment:
Please note that the premise of successfully distributing resource into managed clusters is that it must contain the exactly same namespace as control plane did. Since each cluster has the **`default`** namespace by default, we won't be worry in this case. But suppose we change the namespace in **`basic.yaml`** to be **`multi-cluster`**, we will receive an error: